### PR TITLE
v2: do not return objects from sign() functions anymore. Improve examples.

### PR DESCRIPTION
**Pull request by andreibancioiu**
- Do not return objects from `sign()` functions anymore
 - Improve examples

### CODE REVIEW

File: examples/app.js [hunk 1]
```diff
@@ -1,36 +1,49 @@
 import QRCodeModal from "@walletconnect/qrcode-modal";
 import { WalletConnectProvider } from "../out/walletConnectProvider";
 
-export async function main() {
-    let bridge = "https://bridge.walletconnect.org";
+const BridgeUrl = "https://bridge.walletconnect.org";
```
Comment (StanislavSava) [inline examples/app.js:4]: This should probably be pascalCase, since it's a constant

File: examples/app.js [hunk 2]
```diff
@@ -1,36 +1,49 @@
...
-            console.log("onClientLogin()");
-            await provider.signTransaction(new DummyTransaction());
-        },
-        onClientLogout: async function () {
-            console.log("onClientLogout()");
-        }
-    };
+export class MyApp {
+    constructor() {
+        let callbacks = {
```
Comment (StanislavSava) [inline examples/app.js:8]: These don't seem to be reassigned, probably a constant

**Review by StanislavSava — DISMISSED** [review]

**Review by radumojic — APPROVED** [review]

File: examples/app.js (hunk 1 above)
Comment (andreibancioiu) [reply examples/app.js]: Renamed.

**Review by radumojic — APPROVED** [review]

File: src/walletConnectProvider.ts [hunk 3]
```diff
@@ -162,27 +161,19 @@ export class WalletConnectProvider implements IDappProvider {
...
-            return transactions;
+        if (transactions.length !== signatures.length) {
+            Logger.error("signTransactions: Wallet Connect could not sign the transactions. Invalid signatures.");
+            throw new Error("Wallet Connect could not sign the transactions. Invalid signatures.");
         }
 
-        transactions[0].applySignature(Signature.fromHex(signatures.signature), UserAddress.fromBech32(address));
-        
-        return transactions;
+        transactions.map((transaction, key: number) =>
```
Comment (StanislavSava) [inline src/walletConnectProvider.ts:174]: usually you would use .map to create another array, but here it simply goes over the tx. Wouldn't it be better to use forEach or for of?

**Review by StanislavSava — DISMISSED** [review]

File: src/walletConnectProvider.ts (hunk 3 above)
Comment (andreibancioiu) [reply src/walletConnectProvider.ts]: Fixed.

**Review by StanislavSava — APPROVED** [review]

**Review by radumojic — APPROVED** [review]

**Pull request merged**